extensions/amp-consent/amp-consent.md

```diff
@@ -148,17 +161,92 @@ AMP sends the consent instance ID, a generated user id only for this usage and t
 
-To use `promptIfUnknownForGeoGroup`, a `<amp-geo>` component must be included and properly configured. The `promptIfUnknownForGeoGroup` then accepts a key of a geo group of country codes. More details on how `<amp-geo>` works can be found [here](https://github.com/ampproject/amphtml/blob/master/extensions/amp-geo/amp-geo.md).
 
-In the case that `checkConsentHref` and `promptIfUnknownForGeoGroup` are both defined. The value from `<amp-geo>` will be respected.
 
-#### promptUI
+#### geoOverride
+
+`geoOverride` provides a way to utilize the `<amp-geo>` component to detect user's geo location to assist client side decisions. 
+
+`geoOverride` is a JSON object keyed by geo group codes which are defined in `<amp-geo>` (details [here](https://github.com/ampproject/amphtml/blob/master/extensions/amp-geo/amp-geo.md)). Each geo override should be a valid `<amp-consent>` config object. AMP will take all the values in the corresponding `geoOverride` to override the existing config. 
+
+Two important tips when configuring `amp-geo`:
+
+- All geo groups should be mutually exclusive. The behavior is undetermined if a user falls into multiple geo override.
+- Provide an `unknownGeoGroup` override for users that are failed be be identified by `<amp-geo>`.
+
+Take the following config as an example:
+
+```json
+{
+  "onUpdateHref": "https://example.com/update-consent",
+  "promptUI": "consent-ui",
+  "consentRequired": false,
+
+  "geoOverride": {
+    "geo1": {
+      "consentRequired": true,
+    }, 
+    "geo2": {
+      "checkConsentHref": "https://example.com/check-consent",
+      "consentRequired": "remote",
+    },
+    "unknownGeoGroup": {
+      "checkConsentHref": "https://example.com/check-consent",
+      "consentRequired": true,
+   }
+  }
+}
+```
+
+For users outside `geo1`, `geo2` & `uknown`, the merged config is
+```
+{
+  "onUpdateHref": "https://example.com/update-consent",
+  "promptUI": "consent-ui",
+  "consentRequired": "false",
+}
+```
+`<amp-consent>` does nothing because `"consentRequired": false`.
+
+
+For users in `geo1`, the merged config is
+```
+{
+  "onUpdateHref": "https://example.com/update-consent",
+  "promptUI": "consent-ui",
+  "consentRequired": true,
+}
+```
+The previous consent state is read from local cache as `checkConsentHref` is not specified. Consent will be collected via the specified prompt UI if cache is empty.
+
+
+For users in `geo2`, the merged config is
+```
+{
+  "onUpdateHref": "https://example.com/update-consent",
+  "promptUI": "consent-ui",
+  "checkConsentHref": "https://example.com/check-consent",
+  "consentRequired": "remote",
+}
+```
+AMP will wait for `checkConsentHref` response to decide whether consent is required from the user. If the response contains `consentRequired: true` then AMP will collect consent via the specified prompt UI if consent state is `unknown`. The consent state from server (if returned) takes precedence over the value from local cache.
```

Comment:
While it's covered in geo1 and unknownGeoGroup, we may want to be clear here. That if the cache value is not unknown. AMP will use that value to unblock user this time.

Reply:
oh, forgot about that. seems  a bit unintuitive from the naming.

Reply:
what should the behavior be if `consentRequired: false` and local cached `consentState: rejected`?

Reply:
We would take it as `rejected`. local stored value is always used.
This would apply to EEA users who have rejected to share data, and they travel to out of Europe